cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -1600,6 +1751,13 @@ static void *_umm_realloc( void *ptr, size_t size ) {
 
     c = umm_assimilate_down(c, 0);
 
+    /*
+     * For the ESP8266 interrupts should not be off for more than 10us.
+     * An unprotect/protect around memmove should be safe to do here.
+     * All variables used are on the stack.
+     */
+    UMM_CRITICAL_EXIT(id_realloc);
+
```

Comment:
Only gotcha here is that the cached ummHeapFreeBlocks will be inconsistent should it be read during the memmove.  Probably not worth worrying about.

Reply:
Yea, I thought about that. I am thinking it is not a critical error and it saves a few bytes of IRAM by not addressing a (minor?) transient calculation error.

Reply:
The name is fine by me.  I've seen camelCase, PascalNames, under_scores, and ijk in the codebase.  The IRQ unsafeness, fine by me too.

Reply:
I didn't think about the case of an ISR doing a malloc during this period and didn't realize the assimilate up and down could capture near all of the memory available. (I saw only 56 bytes remaining in my case when I was trying to capture a heap low watermark) Looks to me of the three places I tried to optimize realloc for lock time only one is viable.  :(